AppCypher/EventsManager
# #152801529 Implement JWT authentication

#### What does the PR do?
Support for authentication of user session

#### Description of task to be completed 
- Create user token on access to `/api/v1/users` and `/api/v1/users/login`
- Create middlewares that authenticate user session

#### How should this be manually tested?
- Clone the project repository 
  
  ```$ git clone https://github.com/AppCypher/EventsManager```

- Checkout into the proper branch 

  ```$ git checkout ch-jwt-authentication-152801529```

- This project requires Nodejs and it is expected to run in Nodejs version 6 or higher

- Install project dependencies by running

  ```$ npm install```

- There project bundles task runners that make trying out parts of the project easy to use
- Below is a list of some of the available task runners
    - `server` to start the server 

    - `test:api` to run the api test suite

    - `build` to start webpack build on all server, frontend and test suite code

    - `build:server` to start webpack build on server code
    
    - `build:client:dev` to start webpack build on client code under a dev environment variable
    
    - `build:client:prod` to start webpack build on client code under a production environment variable

- To8 run the server or test suite, the project code needs to be built 

  ```$ npm run build```

#### Background context 
- Authentication and other validations have not yet been implemented

#### Relevant pivotal tracker stories
152801529, 152801542
